### PR TITLE
feat(UIForm): migrate Code widget

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
Code widget was not migrated, but adapted with a wrapper to be used in UIForm.

**What is the chosen solution to this problem?**
* Migrate it
* improve a11y by attaching the label to the textarea

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/UIForm/fields/Code/Code.component.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+export default class Code extends React.Component {
+	constructor(props) {
+		super(props);
+		this.onChange = this.onChange.bind(this);
+		this.onFinish = this.onFinish.bind(this);
+		this.onLoad = this.onLoad.bind(this);
+	}
+
+	onChange(value, event) {
```
Comment (sgangnant) [inline packages/forms/src/UIForm/fields/Code/Code.component.js]: Couldn't you use `mapDispatchToProps` for this ?

**Comment by jsomsanith** [comment]
@sgangnant easy to read right ? :)
@sgendre The `TextareaWidget` defined in `CodeWidget` file doesn't seem to be used, nor exported. Where is this textarea fallback implemented ?

**Comment by sgendre** [comment]
here is the link
https://github.com/Talend/ui/blob/master/packages/forms/src/widgets/CodeWidget/CodeWidget.component.js#L97

**Comment by jsomsanith** [comment]
Oh it's just a warning. I'll add it

File: output/forms.eslint.txt [hunk 2]
```diff
@@ -17,6 +17,10 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/CheckBox/SimpleCheckBox.component.js
   9:4  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
 
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Code/Code.component.js
+   12:1  error    Exporting mutable 'let' binding, use 'const' instead  import/no-mutable-exports
```
Comment (sgendre) [inline output/forms.eslint.txt]: update it please

File: output/forms.eslint.txt [hunk 3]
```diff
@@ -17,6 +17,10 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/CheckBox/SimpleCheckBox.component.js
   9:4  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
 
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Code/Code.component.js
```
Comment (sgendre) [inline output/forms.eslint.txt]: add the es-lint disable appropriate rule

**Pull request merged**